### PR TITLE
feat: add ed25519 utils and Signer classes

### PR DESCRIPTION
**Pull request by adaddeo**
## Motivation

Two small util additions that will constructing messages easier.

## Change Summary

- add `utils/ed25519` to provide `HubResult` convenience functions around `ed` functions
- introduces classes for `EthereumSigner` and `Ed25519Signer` (backwards compatible with previous types but also with a `sign` function)

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] The title of this PR adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] The PR has been tagged with the appropriate change type label(s) (i.e. documentation, feature, bugfix, or chore)

## Additional Context

These came out of the work I was doing on #355. I'm ok to take one, both, or none of the commits, but they seemed useful.

### CODE REVIEW

**Review by pfletcherhill — CHANGES_REQUESTED** [review]
Overall, I think it's a good idea to refactor the signers and crypto utils. The way they're organized now is pretty awkward and largely a relic of a time before flatbuffers. To that end, I think we can make the refactor more dramatic and not worry about backwards compatibility, and I've left some comments and suggestions to that effect. If we're adding these classes, we should use them everywhere rather than continuing to rely on `KeyPair` in many places.

File: app/src/utils/signer/ethereumSigner.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { SignatureScheme } from '@hub/flatbuffers';
+import { ethers } from 'ethers';
+import { signMessageHash } from '~/flatbuffers/utils/eip712';
+import { HubAsyncResult } from '~/utils/hubErrors';
+import { Signer } from './types';
+
+class EthereumSigner implements Signer {
+  /** 20-byte wallet address */
+  public signerKey: string;
+  public scheme = SignatureScheme.Eip712;
```
[the remark this answers is not in the document]
Comment (adaddeo) [reply app/src/utils/signer/ethereumSigner.ts:10]: yes thats better

**Comment by adaddeo** [comment]
@pfletcherhill incorporated your feedback

- `Signers` only need a privateKey, everything else is derived
- `generate*` functions are gone and replaced with `Factories.Eip712Signer` and `Factories.Ed25519Signer`
- `ethers.Wallet` dependency removed from `Factories` and nearly all tests

File: app/src/signers/eip712Signer.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import Signer from './signer';
+
+class Eip712Signer extends Signer {
+  /** 32-byte wallet address */
+  public declare readonly signerKey: Uint8Array;
+  private readonly wallet: Wallet;
+
+  constructor(privateKey: Uint8Array) {
+    const wallet = new Wallet(privateKey);
+    const signerKey = hexStringToBytes(wallet.address.toLowerCase())._unsafeUnwrap();
```
[the remark this answers is not in the document]
Comment (adaddeo) [reply app/src/signers/eip712Signer.ts:16]: Ah! right, I'll open up a new pr with the fix

**Pull request merged**